src/core_ocean/shared/mpas_ocn_mesh.F

```diff
@@ -0,0 +1,842 @@
+                                             cellMaskTmp)
+         call mpas_pool_get_array(meshPool, 'highOrderAdvectionMask',        &
+                                             highOrderAdvectionMaskTmp)
+         call mpas_pool_get_array(meshPool, 'edgeSignOnCell',  &
+                                             edgeSignOnCellTmp)
+         call mpas_pool_get_array(meshPool, 'edgeSignOnVertex',  &
+                                             edgeSignOnVertexTmp)
+         call mpas_pool_get_array(meshPool, 'boundaryEdge',  &
+                                             boundaryEdgeTmp)
+         call mpas_pool_get_array(meshPool, 'boundaryVertex',  &
+                                             boundaryVertexTmp)
+         call mpas_pool_get_array(meshPool, 'boundaryCell',  &
+                                             boundaryCellTmp)
+
+         call mpas_log_write('In meshCreate allocating masks',flushNow=.true.)
+         allocate( &
+            ocnMesh%edgeMask        (ocnMesh%nVertLevels, ocnMesh%nEdges),    &
+            ocnMesh%cellMask        (ocnMesh%nVertLevels, ocnMesh%nCells),    &
+            ocnMesh%vertexMask      (ocnMesh%nVertLevels, ocnMesh%nVertices), &
+            ocnMesh%boundaryEdge    (ocnMesh%nVertLevels, ocnMesh%nEdges),    &
+            ocnMesh%boundaryCell    (ocnMesh%nVertLevels, ocnMesh%nCells),    &
+            ocnMesh%boundaryVertex  (ocnMesh%nVertLevels, ocnMesh%nVertices), &
+            ocnMesh%edgeSignOnCell  (ocnMesh%nVertLevels, ocnMesh%nCells),    &
+            ocnMesh%edgeSignOnVertex(ocnMesh%nVertLevels, ocnMesh%nVertices), &
+            ocnMesh%highOrderAdvectionMask(ocnMesh%nVertLevels, ocnMesh%nEdges))
+
+         call mpas_log_write('In meshCreate converting masks')
+         do n=1,ocnMesh%nCells
+         do k=1,ocnMesh%nVertLevels
+            ocnMesh%cellMask(k,n)       = real(cellMaskTmp(k,n),       RKIND)
+            ocnMesh%edgeSignOnCell(k,n) = real(edgeSignOnCellTmp(k,n), RKIND)
+            ocnMesh%boundaryCell(k,n)   = real(boundaryCellTmp(k,n),   RKIND)
+         end do
+         end do
+
+         do n=1,ocnMesh%nEdges
+         do k=1,ocnMesh%nVertLevels
+            ocnMesh%edgeMask(k,n)     = real(edgeMaskTmp(k,n),     RKIND)
+            ocnMesh%boundaryEdge(k,n) = real(boundaryEdgeTmp(k,n), RKIND)
+            ocnMesh%highOrderAdvectionMask(k,n) = & 
+                              real(highOrderAdvectionMaskTmp(k,n), RKIND)
+         end do
+         end do
+
+         do n=1,ocnMesh%nVertices
+         do k=1,ocnMesh%nVertLevels
+            ocnMesh%vertexMask(k,n)       = real(vertexMaskTmp(k,n),     RKIND)
+            ocnMesh%edgeSignOnVertex(k,n) = &
+                                          real(edgeSignOnVertexTmp(k,n), RKIND)
+            ocnMesh%boundaryVertex(k,n)   = real(boundaryVertexTmp(k,n), RKIND)
+         end do
+         end do
+
+         call mpas_log_write('In meshCreate copy to device')
+#ifdef MPAS_OPENACC
```

Comment:
Normally, I put these data transfers inside an ifdef because we also have an openmp master to make sure we only transfer from one thread in a mixed openmp/openacc setup. But there's an openmp master around most of this routine, so you are correct this can be removed.

Reply:
Just pushed a new version that removed this redundant ifdef